lib/gates_cirq.h

```diff
@@ -0,0 +1,1114 @@
+  static constexpr unsigned num_qubits = 2;
+
+  static GateCirq<fp_type> Create(unsigned time, unsigned q0, unsigned q1,
+                                  fp_type exponent, fp_type global_shift = 0) {
+    fp_type c = std::cos(pi * exponent * 0.5);
+    fp_type s = std::sin(pi * exponent * 0.5);
+    fp_type gc = std::cos(pi * exponent * global_shift);
+    fp_type gs = std::sin(pi * exponent * global_shift);
+    fp_type ec = std::cos(pi * exponent * (0.5 + global_shift));
+    fp_type es = std::sin(pi * exponent * (0.5 + global_shift));
+
+    // Matrix is in this form because the simulator uses inverse qubit order.
+    return CreateGate<GateCirq<fp_type>, CXPowGate>(
+        time, q0, q1, {gc, gs, 0, 0, 0, 0, 0, 0,
+                       0, 0, c * ec, c * es, 0, 0, s * es, -s * ec,
+                       0, 0, 0, 0, gc, gs, 0, 0,
+                       0, 0, s * es, -s * ec, 0, 0, c * ec, c * es},
+        {exponent, global_shift});
+  }
+
+  static schmidt_decomp_type<fp_type> SchmidtDecomp(
+      fp_type exponent, fp_type global_shift) {
+    fp_type c = std::cos(pi * exponent * 0.5);
+    fp_type s = std::sin(pi * exponent * 0.5);
+    fp_type gc = std::cos(pi * exponent * global_shift);
+    fp_type gs = std::sin(pi * exponent * global_shift);
+    fp_type ec = std::cos(pi * exponent * (0.5 + global_shift));
+    fp_type es = std::sin(pi * exponent * (0.5 + global_shift));
+
+    return schmidt_decomp_type<fp_type>{
+      {{1, 0, 0, 0, 0, 0, 0, 0}, {gc, gs, 0, 0, 0, 0, gc, gs}},
+      {{0, 0, 0, 0, 0, 0, 1, 0}, {c * ec, c * es, s * es, -s * ec,
+                                  s * es, -s * ec, c * ec, c * es}},
+    };
+  }
+};
+
+// The (exponent=phi/pi, global_shift=-0.5) instance of XPowGate.
+// This is a function in Cirq.
+template <typename fp_type>
+struct rx {
+  static constexpr GateKind kind = krx;
+  static constexpr char name[] = "rx";
+  static constexpr unsigned num_qubits = 1;
+
+  static GateCirq<fp_type> Create(unsigned time, unsigned q0, fp_type phi) {
+    return XPowGate<fp_type>::Create(time, q0, inv_pi * phi, -0.5);
+  }
+};
+
+// The (exponent=phi/pi, global_shift=-0.5) instance of YPowGate.
+// This is a function in Cirq.
+template <typename fp_type>
+struct ry {
+  static constexpr GateKind kind = krx;
+  static constexpr char name[] = "ry";
+  static constexpr unsigned num_qubits = 1;
+
+  static GateCirq<fp_type> Create(unsigned time, unsigned q0, fp_type phi) {
+    return YPowGate<fp_type>::Create(time, q0, inv_pi * phi, -0.5);
+  }
+};
+
+// The (exponent=phi/pi, global_shift=-0.5) instance of ZPowGate.
+// This is a function in Cirq.
+template <typename fp_type>
+struct rz {
+  static constexpr GateKind kind = krx;
+  static constexpr char name[] = "rz";
+  static constexpr unsigned num_qubits = 1;
+
+  static GateCirq<fp_type> Create(unsigned time, unsigned q0, fp_type phi) {
+    return ZPowGate<fp_type>::Create(time, q0, inv_pi * phi, -0.5);
+  }
+};
```

Comment:
It looks like these (rx/ry/rz) aren't separate from the X/Y/ZPowGates - the gates returned by their `Create()` methods will have the properties of the *PowGate structs.

A few options for handling this:
- Remove these entirely, replacing references with *PowGate.
- Remove the member variables in these (to avoid confusion).
- Promote these to separate gates, similar to `riswap` vs. ISWAPPowGate.

Reply:
Done.